### PR TITLE
Employee pages

### PR DESCRIPTION
**Pull request by Summer-luna**
# Description

> Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
- display employee list ui
- get employee data from database via graphql and display on employee list page
- add employee ui
- able to add employee to database via employee ui
## Checklist

- [x] This PR can be reviewed in under 30 minutes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have assigned reviewers to this PR.

### CODE REVIEW

File: packages/client/src/.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#VITE_BACKEND_URL=https://test-time-backend.sail.codes
```
Comment (gregfrasco) [inline packages/client/src/.env:1]: .env file shouldn't be commited

File: packages/client/src/.env.example [hunk 2]
```diff
@@ -1 +0,0 @@
-VITE_BACKEND_URL=https://test-time-backend.sail.codes
```
Comment (gregfrasco) [inline packages/client/src/.env.example:1]: Restore this file

File: packages/client/src/graphql/graphql-provider.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React, { FC } from 'react';
 import { ApolloClient, ApolloProvider, from, HttpLink, InMemoryCache } from '@apollo/client';
-
-const uri = `${import.meta.env.VITE_BACKEND_URL}/graphql`;
+// import.meta.env.VITE_BACKEND_URL
```
Comment (gregfrasco) [inline packages/client/src/graphql/graphql-provider.tsx:3]: fix this hardcoding

File: packages/client/src/main.tsx [hunk 4]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import App from './App';
+import { GraphqlProvider } from '@graphql/graphql-provider';
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
   <React.StrictMode>
-    <App />
+    <GraphqlProvider>
```
Comment (gregfrasco) [inline packages/client/src/main.tsx:8]: This should be located inside app.tsx

File: packages/client/src/pages/Employee/addEmployee.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { EmployeeForm } from '@pages/Employee/components/form/EmployeeForm';
+
+export const AddEmployee = () => {
+  return (
+    <div>
```
Comment (gregfrasco) [inline packages/client/src/pages/Employee/addEmployee.tsx:5]: this div and component seems to be unessarary

File: packages/client/src/pages/Track/components/DisplayCard.component.tsx [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import Card from '@mui/material/Card';
 import CardContent from '@mui/material/CardContent';
 import Typography from '@mui/material/Typography';
-import { WorkSuitIcon } from '@components/images/WorkSuitIcon';
-import { RestIcon } from '@components/images/RestIcon';
+import { WorkSuitIcon } from '@pages/Track/components/images/WorkSuitIcon';
```
Comment (gregfrasco) [inline packages/client/src/pages/Track/components/DisplayCard.component.tsx:4]: It seems like these can stay in the components folders

File: packages/server/package.json [hunk 7]
```diff
@@ -35,6 +35,7 @@
     "@nestjs/platform-express": "^9.0.0",
     "@prisma/client": "4.10.1",
     "apollo-server-express": "^3.11.1",
+    "class-transformer": "^0.5.1",
```
Comment (gregfrasco) [inline packages/server/package.json:38]: Is this package used?

File: packages/client/src/pages/Employee/components/form/EmployeeForm.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      // clean the form values
+      resetForm();
+    }
+  });
+
+  if (error) return <Container sx={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100vh' }}>`Submission error! ${error.message}`</Container>;
+
+  return (
+    <Container>
+      <form onSubmit={formik.handleSubmit}>
```
Comment (gregfrasco) [inline packages/client/src/pages/Employee/components/form/EmployeeForm.tsx:41]: Use formik Form component

File: packages/client/src/pages/Employee/components/form/EmployeeForm.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        variables: {
+          newEmployee: { ...values, rate: parseFloat(values.rate) }
+        }
+      });
+      // clean the form values
+      resetForm();
+    }
+  });
+
+  if (error) return <Container sx={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100vh' }}>`Submission error! ${error.message}`</Container>;
```
Comment (gregfrasco) [inline packages/client/src/pages/Employee/components/form/EmployeeForm.tsx:37]: Dont use 1 line if statements

File: packages/client/src/pages/Employee/components/form/EmployeeForm.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+  return (
+    <Container>
+      <form onSubmit={formik.handleSubmit}>
+        <Box sx={{ display: 'flex', flexDirection: 'column', gap: '1rem' }}>
+          <TextField id="name" name="name" type="text" placeholder="Employee Name" value={formik.values.name} onChange={formik.handleChange} required />
+          <TextField
```
Comment (gregfrasco) [inline packages/client/src/pages/Employee/components/form/EmployeeForm.tsx:44]: Probably should bring in these components to simply these inputs

https://github.com/hicsail/authentication-service/blob/main/packages/client/src/components/forms/text-input.tsx

**Review by gregfrasco — CHANGES_REQUESTED** [review]

File: packages/server/package.json (hunk 7 above)
Comment (Summer-luna) [reply packages/server/package.json:38]: no, but when I was trying to run nest start, it gave an error and asked me to install this package. But right now I tried to uninstall it, it works fine.

**Pull request closed without merging**